json/decode.go

```diff
@@ -617,11 +618,19 @@ func (d *decodeState) object(v reflect.Value) {
 		start := d.off - 1
 		op = d.scanWhile(scanContinue)
 		item := d.data[start : d.off-1]
-		key, ok := unquoteBytes(item)
+		key, ok := unquote(item)
```

Comment:
It was previously calling `unquoteBytes` so it could do case-folding (by calling `equalFold` below), but that's not relevant anymore as I deleted all the case-folding logic. Calling `unquote` is equivalent to `unquoteBytes`, except that the return value is a string instead of the raw bytes of the key. Changed it because maps don't allow byte slice indices (because of their mutability I believe).